prov/sockets/src/sock_av.c

```diff
@@ -618,10 +628,12 @@ int sock_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 	*av = &_av->av_fid;
 	return 0;
 
+#ifndef _WIN32
 err_shm:
 	shm_unlink(_av->name);
 	ret = -FI_EINVAL;
 	goto err2;
+#endif /* _WIN32 */
 err3:
 	free(_av->table_hdr);
 err2:
```

Comment:
This may need to be restructured to avoid the ifdefs.  Have I mentioned how much I dislike ifdefs?  :)

Can the missing definitions (shm_unlink, etc.) just be added to windows/osd.h with a return value of -1?